tests/test_addons.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+from typing import Any, Dict, Optional
+
+from scrapy.crawler import Crawler
+from scrapy.settings import BaseSettings
+from scrapy.utils.test import get_crawler
+
+
+class GoodAddon:
+    def __init__(self, config: Optional[Dict[str, Any]] = None) -> None:
+        super().__init__()
+        self.config = config or {}
+
+    def update_settings(self, settings):
+        settings.update(self.config, "addon")
+
+
+class CreateInstanceAddon:
+    def __init__(self, crawler: Crawler) -> None:
+        super().__init__()
+        self.crawler = crawler
+        self.config = crawler.settings.getdict("MYADDON")
+
+    @classmethod
+    def from_crawler(cls, crawler: Crawler):
+        return cls(crawler)
+
+    def update_settings(self, settings):
+        settings.update(self.config, "addon")
+
+
+class AddonTest(unittest.TestCase):
+    def test_update_settings(self):
+        settings = BaseSettings()
+        settings.set("KEY1", "default", priority="default")
+        settings.set("KEY2", "project", priority="project")
+        addon_config = {"KEY1": "addon", "KEY2": "addon", "KEY3": "addon"}
+        testaddon = GoodAddon(addon_config)
+        testaddon.update_settings(settings)
+        self.assertEqual(settings["KEY1"], "addon")
+        self.assertEqual(settings["KEY2"], "project")
+        self.assertEqual(settings["KEY3"], "addon")
+
+
+class AddonManagerTest(unittest.TestCase):
```

Comment:
Could you please add a couple more tests:

1. Check that addon priority works (i.e. a right addon is used , based on the priority)
2. Check that using addon classes works in settings, not just paths

Something else?

Reply:
I think we should also add tests for the suggestions we make in the documentation, e.g. using "addon" priority and they way it interacts with other settings; an addon which reads the default setting and uses a fallback implementation.